docs/deployment/feature_gates.md

```diff
@@ -189,14 +190,15 @@ A *General Availability* (GA) feature is also referred to as a *stable* feature.
+| IPv6SingleStack                     | `gardener-apiserver`, `gardenlet` | Allows creating seed and shoot clusters with [IPv6 single-stack networking](../usage/ipv6.md) enabled in their spec ([GEP-21](../proposals/21-ipv6-singlestack-local.md)). If enabled in gardenlet, the default behavior is unchanged, but setting `ipFamilies=[IPv6]` in the `seedConfig` is allowed. Only if the `ipFamilies` setting is changed, gardenlet behaves differently. |
+| MutableShootSpecNetworkingNodes     | `gardener-apiserver`              | Allows updating the field `spec.networking.nodes`. The validity of the values has to be checked in the provider extensions. Only enable this feature gate when your system runs provider extensions which have implemented the validation.                                                                                                                                         |
+| ShootForceDeletion                  | `gardener-apiserver`              | Allows forceful deletion of Shoots by annotating them with the `confirmation.gardener.cloud/force-deletion` annotation.                                                                                                                                                                                                                                                            |
+| UseNamespacedCloudProfile           | `gardener-apiserver`              | Enables usage of `NamespacedCloudProfile`s in `Shoot`s.                                                                                                                                                                                                                                                                                                                            |
+| ShootManagedIssuer                  | `gardenlet`                       | Enables the shoot managed issuer functionality described in GEP 24.                                                                                                                                                                                                                                                                                                                |
+| BilinearPodAutoscalingForAPIServer  | `gardenlet`                       | When enabled, shoot kube-apiserver is scaled simultaneously via HPA on request rate and VPA on resource usage. When enabled, this feature takes prececence over the HVPA and HVPAForShootedSeed features in determining how shoot kube-apiserver pods are scaled.                                                                                                                  |
```

Comment:
Yes, we are aware of it. However, currently in many places https://github.com/gardener/gardener-custom-metrics assumes that it runs in a Seed cluster and it scrapes kube-apiserver Pods in Shoot namespaces.
IMO, it would need a redesign/refactoring of https://github.com/gardener/gardener-custom-metrics to make it support configurable scrape targets or something similar so that it handles both scenarios - seed cluster and garden runtime cluster.
We would like to first try to cover the scenario for the Shoot kube-apiservers and to make sure that GEP-23 really works - brings the hoped stability and utilization improvements. We are aware that we have to make it work also for the virtual-kube-apiserver and gardener-apiserver as well to eliminate the HVPA usages there as well.
If you don't have objections, we would like to leave it out of scope for now for the first milestone (make sure that it works for Shoot kube-apiservers).

Reply:
OK, that's unfortunate, but it is what it is. Then let's focus on the shoot scenario for now - perhaps you could add a task for the garden runtime cluster to https://github.com/gardener/gardener/issues/8259.

Reply:
Addition: I guess, it still makes sense to already make this gcmx component generic enough so that it does work with both scenarios (i.e., prevent usage of seed/shoot, etc., see other comments below).

Reply:
Just to be clear, this is not an omission, but an intentional phased approach. The garden runtime case is [an explicit non-goal](https://github.com/gardener/gardener/blob/master/docs/proposals/23-autoscaling-kube-apiserver-via-independent-hpa-and-vpa.md#non-goals) for the current phase.